README.md

```diff
@@ -159,62 +169,209 @@ extension MyViewController: UITableViewDataSource {
+>   // of the variable `cell` below is infered to be `ParentCell` too. So only methods
+>   // declared in the parent `ParentCell` class will be accessible on the `cell` variable.
+>   let cell = tableView.dequeueReusableCell(indexPath: indexPath, cellType: cellClass)
+>   return cell  
+> }
+> ```
+> </details>
+
+---
+
+
 
-## Instantiating arbitrary views from XIBs
+# Type-safe XIB-based reusable views
 
-If you mark an arbitrary (non-cell) `UIView` as `NibLoadable` [as demonstrated above](#example-with-a-xib-based-arbitrary-uiview), you can also instantiate such a view using its Nib by calling `loadFromNib()`:
+`Reusable` also allows you to create reusable custom views designed in Interface Builder to reuse them in other XIBs or by code, like creating custom UI widgets used in multiple places in your app.
+
+## 1. Declare your views to conform to `NibLoadable` or `NibOwnerLoadable`
+
+In your swift source declaring your custom view class:
+
+* Use the `NibLoadable` protocol if the XIB you're using don't use its "File's Owner" and the reusable view you're designing is the root view of the XIB
+* Use the `NibOwnerLoadable` protocol if you used a "File's Owner" of the XIB being of the class of your reusable view, and the root view(s) of the XIB is to be set as a subview providing its content.
 
 ```swift
-let instance1 = NibBasedRandomView.loadFromNib()
-let instance2 = NibBasedRandomView.loadFromNib()
-let instance3 = NibBasedRandomView.loadFromNib()
-…
+// a XIB-based custom UIView, used as root of the XIB
+final class NibBasedRootView: UIView, NibLoadable { /* and that's it! */ }
+
+// a XIB-based custom UIView, used as the XIB's "File's Owner"
+final class NibBasedFileOwnerView: UIView, NibOwnerLoadable { /* and that's it! */ }
 ```
 
-## Instanciating ViewControllers from Storyboards 
+> 💡 You should use the second approach if you plan to use your custom view in another XIB or Storyboard.  
+> This will allow you to just drop a UIView in a XIB/Storyboard and change its class to the class of your custom XIB-based view to use it. That custom view will then automagically load its own content from the associated XIB when instantiated by the storyboard containing it, without having to write additional code to load the content of the ustom view manually every time.
+
+## 2. Design your view in Interface Builder
 
-### Initial ViewController
+For example if you named your class `MyCustomView` and made it `NibOwnerLoadable`:
 
-If one of your custom `UIViewController` (named `CustomVC` for example) is **designed as the initial ViewController of a Storyboard** (named `CustomVC.storyboard`):
+* Set the _File's Owner_'s class to `MyCustomView`
+* Design the content of the view via the root view of that XIB (which is a standard `UIView` with no custom class) and its subviews
+* Connect any `@IBOutlets` and `@IBActions` between the _File's Owner_ (the `MyCustomView`) and its content
```

Comment:
The screenshot uses `MyCustomWidget` but the text uses `MyCustomView`, should try to be consistent.